tiledb/sm/compressors/dd_compressor.cc

```diff
@@ -71,17 +69,29 @@ Status DoubleDelta::compress(
       return DoubleDelta::compress<int64_t>(input_buffer, output_buffer);
     case Datatype::UINT64:
       return DoubleDelta::compress<uint64_t>(input_buffer, output_buffer);
-    default:
-      return LOG_STATUS(Status::TileIOError(
-          "Cannot compress tile with DoubleDelta; Not supported datatype"));
+    case Datatype::CHAR:
+    case Datatype::STRING_ASCII:
+    case Datatype::STRING_UTF8:
+    case Datatype::STRING_UTF16:
+    case Datatype::STRING_UTF32:
+    case Datatype::STRING_UCS2:
+    case Datatype::STRING_UCS4:
+      return DoubleDelta::compress<char>(input_buffer, output_buffer);
```

Comment:
Do we have to be careful here w.r.t. `char` meaning different things on different compilers? E.g. is it safer to treat the buffer as `uint8_t` here?